ui/src/pages/version/HistoryDetails.js

```diff
@@ -14,48 +14,41 @@ import { useMerlinApi } from "../../hooks/useMerlinApi";
 
 const defaultTextSize = "s";
 
-const DeploymentStatus = ({
-  status,
-  deployment,
-  endpointRevisionId,
-  endpointStatus,
-}) => {
-  if (status === "running" || status === "serving") {
-    if (
-      deployment.revision_id === endpointRevisionId &&
-      (endpointStatus === "pending" ||
-        endpointStatus === "running" ||
-        endpointStatus === "serving")
-    ) {
+const DeploymentStatus = ({ status }) => {
+  switch (status) {
+    case "pending":
+      return <EuiHealth color="gray">Pending</EuiHealth>;
+    case "running":
+    case "serving":
       return <EuiHealth color="success">Deployed</EuiHealth>;
-    }
-    return <EuiHealth color="default">Not Deployed</EuiHealth>;
-  } else if (status === "pending") {
-    return <EuiHealth color="gray">Pending</EuiHealth>;
-  }
-
-  if (deployment.error !== "") {
-    return <EuiHealth color="danger">Failed</EuiHealth>;
+    case "terminated":
+      return <EuiHealth color="default">Not Deployed</EuiHealth>;
+    case "failed":
+      return <EuiHealth color="danger">Failed</EuiHealth>;
+    default:
+      return <EuiHealth color="subdued">-</EuiHealth>;
   }
 };
 
 const RevisionPanel = ({ deployments, deploymentsLoaded, endpoint }) => {
   const [orderedDeployments, setOrderedDeployments] = useState([]);
+  const [currentDeployment, setCurrentDeployment] = useState({ id: null });
+
   useEffect(() => {
-    let orderedDeployments = deployments.sort((a, b) => (a.id > b.id ? 1 : -1));
-
-    let idx = 0;
-    orderedDeployments.forEach((deployment) => {
-      if (deployment.status === "running" || deployment.status === "serving") {
-        deployment.revision_id = idx + 1;
-        idx++;
-      }
+    const ordered = deployments.sort((a, b) => (a.id < b.id ? 1 : -1));
+    setOrderedDeployments(ordered);
+
+    const currentDeployment = ordered.find((deployment) => {
+      return (
+        (deployment.status === "running" ||
+          deployment.status === "serving" ||
+          deployment.status === "terminated") &&
+        deployment.error === ""
+      );
```

Comment:
We should also handle the case where all previous deployment attempts were failed (`currentDeployment` is empty) - in this case, the latest record is simply the current.

Reply:
Updated:

<img width="500" alt="image" src="https://github.com/caraml-dev/merlin/assets/8122852/910c6fc2-fffd-4f54-9391-40ef94bf0b03">